mlir/lib/Target/SPIRV/Deserialization/Deserializer.cpp

```diff
@@ -867,11 +867,13 @@ LogicalResult spirv::Deserializer::processType(spirv::Opcode opcode,
     typeMap[operands[0]] = IntegerType::get(context, operands[1], sign);
   } break;
   case spirv::Opcode::OpTypeFloat: {
-    if (operands.size() != 2)
-      return emitError(unknownLoc, "OpTypeFloat must have bitwidth parameter");
+    if (operands.size() != 2 && operands.size() != 3)
+      return emitError(unknownLoc, "OpTypeFloat must have bitwidth parameter "
+                                   "and optional floating point encoding");
```

Comment:
This error message is a bit confusing to me and reads like the optional encoding is mandatory. Could you reword this a little bit?